EHR_Purchasing/src/org/labkey/ehr_purchasing/EHR_PurchasingContainerListener.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.labkey.ehr_purchasing;
+
+import org.jetbrains.annotations.NotNull;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager.ContainerListener;
+import org.labkey.api.data.DatabaseTableType;
+import org.labkey.api.data.DbScope;
+import org.labkey.api.data.SimpleFilter;
+import org.labkey.api.data.Table;
+import org.labkey.api.data.TableInfo;
+import org.labkey.api.security.User;
+import java.util.Collections;
+import java.util.Collection;
+
+import java.beans.PropertyChangeEvent;
+
+public class EHR_PurchasingContainerListener implements ContainerListener
+{
+    @Override
+    public void containerCreated(Container c, User user)
+    {
+    }
+
+    @Override
+    public void containerDeleted(Container c, User user)
+    {
+        // This will clean up the ehr_purchasing schema.  For extensible tables, exp module should clean up related exp data.
+        DbScope scope = EHR_PurchasingSchema.getInstance().getSchema().getScope();
+        SimpleFilter containerFilter = SimpleFilter.createContainerFilter(c);
+        try (DbScope.Transaction transaction = scope.ensureTransaction())
+        {
+            TableInfo vendorTable = EHR_PurchasingSchema.getInstance().getVendorTable();
+            if (vendorTable.getTableType() == DatabaseTableType.TABLE)
```

Comment:
Why this check? Shouldn't it always be a table in the DB?

Reply:
removed